### PR TITLE
Check for actual usage of java.lang.Void in ForbiddenVoid rule

### LINKED ISSUE #1642: False positive on ForbiddenVoid
## Observed Behavior
False positive on ForbiddenVoid when theres a property/field/enum/class name/class or something called void.

## Steps to Reproduce
Call a class Void, o create a Void value in a enum.

## Your Environment

* Version of detekt used: RC 14, last one I think.
* Version of Gradle used (if applicable):
* Operating System and version:
* Link to your project (if it's a public repository):

### PR DESCRIPTION
**Pull request by 3flex**
Rule now requires type resolution, but should not trigger false positives anymore.

Fixes #1642

### CODE REVIEW

**Comment by 3flex** [comment]
These OOM errors are a bit concerning, one each on macOS and Windows while tests are running... but they are not from this PR, there were a couple recently in CI.

Will merge but something to keep an eye on.

**Comment by schalkms** [comment]
> java.lang.OutOfMemoryError: GC overhead limit exceeded
>org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.access$contentRootToVirtualFile(KotlinCoreEnvironment.kt:125)
>java.lang.OutOfMemoryError: Java heap space

The more code we `compileAndLint` with the help of Jsr223, the more memory the compile engine needs.
@3flex I investigated this issue and came to the following conclusion.
In the combination with some sort of special code, especially with the `UselessCallOnNotNull` rule tests, the compile engine throws OOM errors that are concerning.
In that case, I think this engine isn't suitable for heavy workloads like we have.
As mentioned, there is special code that utilizes the full capacity of the Jsr223 compile engine (for instance `UselessCallOnNotNull` rule tests).
Anyways, I prepared a PR in order to mitigate this problems.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 import org.spekframework.spek2.style.specification.describe
 
 class ForbiddenVoidSpec : Spek({
     val subject by memoized { ForbiddenVoid(Config.empty) }
 
+    lateinit var environment: KotlinCoreEnvironment
+
     describe("ForbiddenVoid rule") {
+        beforeEachTest {
+            environment = KtTestCompiler.createEnvironment()
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: I think this line causes OOM, I think we need to dispose the kotlin environment

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: You might be right. I'm currently working on a fix.

**Pull request merged**